virtual_ecosystem/models/animal/animal_model.py

```diff
@@ -744,23 +749,23 @@ def remove_dead_cohort(self, cohort: AnimalCohort) -> None:
         Raises:
             KeyError: If the cohort ID does not exist in the model's cohorts.
         """
-        # Check if the cohort exists in self.cohorts
-        if cohort.id in self.cohorts:
+        # Check if the cohort exists in self.active_cohorts
+        if cohort.id in self.active_cohorts:
             # Iterate over all grid cell keys in the cohort's territory
             for cell_id in cohort.territory:
                 if cell_id in self.communities and cohort in self.communities[cell_id]:
                     self.communities[cell_id].remove(cohort)
 
             # Remove the cohort from the model's cohorts dictionary
-            del self.cohorts[cohort.id]
+            del self.active_cohorts[cohort.id]
         else:
             raise KeyError(f"Cohort with ID {cohort.id} does not exist.")
 
     def remove_dead_cohort_community(self) -> None:
         """This handles remove_dead_cohort for all cohorts in a community."""
         # Collect cohorts to remove (to avoid modifying the dictionary during iteration)
         cohorts_to_remove = [
-            cohort for cohort in self.cohorts.values() if cohort.individuals <= 0
+            cohort for cohort in self.active_cohorts.values() if cohort.individuals <= 0
```

Comment:
I'd rewrite the code a bit so that `cohort.individuals` can never be <0.

Reply:
I just checked and it actually can't slip below zero. I'm realizing that I've got a defensive coding habit that I should probably chill out about!

Reply:
> I just checked and it actually can't slip below zero. I'm realizing that I've got a defensive coding habit that I should probably chill out about!

You can also go ultra-pythonic and use "truthiness" rather than `== 0` 😄 

```python
In [314]: @dataclass
     ...: class Test():
     ...:     a: int
     ...: 

In [315]: x = Test(a=1)

In [316]: z = Test(a=0)

In [317]: [z for z in [x, z] if not z.a]
Out[317]: [Test(a=0)]
```

I don't know if that is as clear though!

Reply:
Although that _doesn't_ catch values below zero.